### DISCUSSION TITLE
InvokeCommandAction on listbox items

### DISCUSSION THREAD

**Discussion by GuilhermeRaposo** [AvaloniaUI/Avalonia.Xaml.Behaviors#147] [General]
How do I invoke a command when a listbox item is double clicked?

```
<ListBox Items="{ Binding Accounts }" SelectedItem="{ Binding SelectedAccount }" ScrollViewer.VerticalScrollBarVisibility="Visible" Grid.Row="2">
    <ListBox.ItemTemplate>
        <DataTemplate>
            <TextBlock x:Name="{ Binding AccountName }" Text="{ Binding AccountName }">
                <i:Interaction.Behaviors>
                    <ia:EventTriggerBehavior EventName="DoubleTapped" SourceObject="{ Binding AccountName }">
                        <ia:InvokeCommandAction Command="{ Binding Login }" />
                    </ia:EventTriggerBehavior>
                </i:Interaction.Behaviors>
            </TextBlock>
        </DataTemplate>
    </ListBox.ItemTemplate>
</ListBox>
```

This doesn't seem to work, but if I wrap the behavior inside a Button and change the SourceObject to the button it works normally.

**Comment by BradBergstrom** [comment]
I am dealing with the same issue, did you find a different solution other than adding a button to the ListBoxItem?

**Reply by GuilhermeRaposo** [reply]
Sorry for not being able to help that much, it's been a while since I worked on this, but I believe some kind soul from the Avalonia telegram chat (not sure if that's still a thing) helped me fix the issue like [this](https://github.com/GuilhermeRaposo/Steam-Account-Switcher/blob/a844ca4cc3dcbc166f38cb736686b549d1c47571/SteamAccountSwitcher/Views/AccountsView.axaml#L41C1-L45C33).

I don't really recall if that fixed the issue or if it was the Grid wrap around it, but it did work.

**Reply by BradBergstrom** [reply]
You are a saint. That did the trick, thanks!
